### PR TITLE
fix(compat): drop Buffer and TextDecoder for browser/React Native support

### PR DESCRIPTION
**Pull request by Sednaoui**
generateOnChainIdentifier used Buffer.from and ABI string decoding used TextDecoder, both undefined in browsers (no polyfill) and React Native/Hermes. Replaced with the existing pure-JS toUtf8Bytes plus a new fromUtf8Bytes decoder, and moved the WebAuthn adapter and Calibur executor decode off TextEncoder/TextDecoder. Verified in headless Chrome, real Hermes, a real Android device (passkeys + live sponsored tx), and a real browser passkey sponsored tx.

## Summary by CodeRabbit

* **New Features**
  * Added and exported pure‑JS UTF‑8 encode/decode utilities.

* **Refactor**
  * Switched UTF‑8 handling across account modules to the new pure‑JS utilities for broader runtime compatibility (e.g., React Native/Hermes).

* **Bug Fixes**
  * Preserve raw binary payloads when encoding/decoding batched call data to avoid corruption.

* **Tests**
  * Added thorough tests covering strict UTF‑8 decoding edge cases and regression tests for batched calldata preservation.

### CODE REVIEW

File: src/ethereUtils.ts [hunk 1]
```diff
@@ -339,6 +339,36 @@ function toUtf8Bytes(str: string): Uint8Array {
...
+export function fromUtf8Bytes(bytes: Uint8Array): string {
+    let result = "";
+    let i = 0;
+    while (i < bytes.length) {
+        const b0 = bytes[i++];
+        if (b0 < 0x80) {
+            result += String.fromCharCode(b0);
+        } else if (b0 >= 0xc0 && b0 < 0xe0) {
+            const b1 = bytes[i++] & 0x3f;
+            result += String.fromCharCode(((b0 & 0x1f) << 6) | b1);
```
[the remark this answers is not in the document]
Comment (Sednaoui) [reply src/ethereUtils.ts]: fixed
